.github/workflows/builNTest.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build and Test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build-and-test-on-all-OS:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
```

Comment:
We had discussed this earlier but I think it may be good to bring up with @psybers and @esherman77 : while I'm open to transitioning to a newer version of Java now that SPF is no longer a required dependency, SV-COMP currently notes in their repo to assume all tests are written in Java 8, so changing to a newer version may mean taking additional caution when converting files (e.g. rejecting files or individual nodes using methods/types added to the JDK in later versions)

I would note the ASTParser allows us to specify JLS8 when parsing a file even if we are running in a newer version of Java.

Reply:
You can technically use newer versions, but we would need a JDK 8 runtime to make sure everything really complies on JDK 8.  So it is just easier to stick to JDK 8, unless we absolutely need a newer one for one or more of our libraries.